eurocropsml/acquisition/clipping/clipper.py

```diff
@@ -230,83 +239,101 @@ def clipping(
-    df_final.set_index(config.parcel_id_name, inplace=True)
-
-    new_data: bool = False
-    if processed < len(args):
-        new_data = True
-        logger.info("Starting parallel raster clipping...")
-        te = tqdm(total=len(args) - processed, desc="Clipping raster tiles.")
-        while processed < len(args):
-            chunk_args: list[tuple[pd.DataFrame, list]] = args[processed : processed + chunk_size]
-            results: list[pd.DataFrame] = []
-
-            with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
-                futures = [executor.submit(func, *arg) for arg in chunk_args]
-
-                for future in concurrent.futures.as_completed(futures):
-                    try:
-                        result = future.result()
-                        results.append(result)
-                    except Exception:
-                        results.append(None)
-
-            # Process and save results
-            for result in results:
-                if result is not None and not result.empty:
-                    df_final = df_final.fillna(result)
-                te.update(n=1)
-
-            processed += len(chunk_args)
-            if processed % save_files == 0:
-                df_final.to_pickle(clipped_dir.joinpath(f"Final_{file_counts}.pkg"))
-                del df_final
-                df_final = polygon_df.copy()
-                df_final.set_index(config.parcel_id_name, inplace=True)
-                file_counts += 1
-            # Clear variables to release memory
-            del chunk_args, futures
-            gc.collect()
-
-        df_final.to_pickle(clipped_dir.joinpath(f"Final_{file_counts}.pkg"))
-
-    _merge_dataframe(
-        polygon_df,
-        clipped_dir,
-        clipping_path,
-        cast(str, config.parcel_id_name),
-        new_data,
-    )
+        polygon_df_month = polygon_df_month.drop(["geometry"], axis=1)
+        df_final_month = polygon_df_month.copy()
+        df_final_month.set_index(config.parcel_id_name, inplace=True)
+
+        new_data: bool = False
+        if processed < len(args_month):
+            new_data = True
+            te = tqdm(
+                total=len(args_month) - processed, desc=f"Clipping raster tiles for {month_name}..."
+            )
+            while processed < len(args_month):
+                chunk_args: list[tuple[pd.DataFrame, list]] = args_month[
+                    processed : processed + chunk_size
+                ]
+                results: list[pd.DataFrame] = []
+
+                with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+                    futures = [executor.submit(func, *arg) for arg in chunk_args]
+
+                    for future in concurrent.futures.as_completed(futures):
+                        try:
+                            result = future.result()
+                            results.append(result)
+                        except Exception:
```

Comment:
What type of Exception are we catching here?

Reply:
Replace Exception by AttributeError.